src/Orleans.Core/Transactions/ITransactionParticipant.cs

```diff
@@ -0,0 +1,136 @@
+        /// <param name="totalParticipants">the total number of participants in the transaction</param>
+        /// <returns>the status of the transaction</returns>
+        Task<Status> PrepareAndCommit(Guid transactionId, AccessCounter accessCount, DateTime timeStamp,
+            List<ITransactionParticipant> writeParticipants, int totalParticipants);
+
+        /// <summary>
+        /// One-way message sent by TA to participants to let them know a transaction has aborted.
+        /// </summary>
+        /// <param name="transactionId">The id of the aborted transaction</param>
+        Task Abort(Guid transactionId);
+
+        /// <summary>
+        /// One-way message sent by a participant to the TM after it (successfully or unsuccessfully) prepares.
+        /// </summary>
+        /// <param name="transactionId">The id of the transaction</param>
+        /// <param name="timeStamp">The commit timestamp of the transaction</param>
+        /// <param name="participant">The participant sending the message</param>
+        /// <param name="Status">The outcome of the prepare</param>
+        /// <returns></returns>
+        Task Prepared(Guid transactionId, DateTime timeStamp, ITransactionParticipant participant, Status status);
+
+        /// <summary>
+        /// One-way message sent by TM to participants to let them know a transaction has aborted.
+        /// </summary>
+        /// <param name="transactionId">The id of the aborted transaction</param>
+        /// <param name="timeStamp">The commit timestamp of the aborted transaction</param>
+        /// <param name="status">Reason for abort</param>
+        Task Cancel(Guid transactionId, DateTime timeStamp, Status status);
+
+        /// <summary>
+        /// Request sent by TM to participants to let them know a transaction has committed.
+        /// Participants respond after cleaning up all prepare records.
+        /// </summary>
+        /// <param name="transactionId">The id of the aborted transaction</param>
+        /// <param name="timeStamp">The commit timestamp of the aborted transaction</param>
+        Task Confirm(Guid transactionId, DateTime timeStamp);
+
+        /// <summary>
+        /// One-way message sent by participants to TM, to let TM know they are still waiting to hear about
+        /// the fate of a transaction.
+        /// </summary>
+        /// <param name="transactionId">The id of the transaction</param>
+        /// <param name="timeStamp">The commit timestamp of the transaction</param>
+        /// <param name="participant">The participant sending the message</param>
+        Task Ping(Guid transactionId, DateTime timeStamp, ITransactionParticipant participant);
+    }
+
+
+    /// <summary>
+    /// Used to propagate information about the status of a transaction. Used for transaction orchestration, for diagnostics, 
+    /// and for generating informative user exceptions
+    /// </summary>
+    public enum Status
```

Comment:
Can we name this something more specific?  TransactionStatus? ..?

Reply:
Sure. Currently fighting with VS to let me rename.

Reply:
Ok, done.